src/main/java/com/example/demo/utils/RouteUtils.java

```diff
@@ -0,0 +1,51 @@
+package com.example.demo.utils;
+
+import com.example.demo.model.City;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ResourceUtils;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class RouteUtils {
+    private List<City> cityList = new ArrayList<>();
```

Comment:
This can be store in Map<String, List<City>> cityMap, so that same name cities can be grouped together and retrieval should be faster